client/core/core.go

```diff
@@ -2632,7 +2640,16 @@ func (c *Core) registerLegacy(dc *dexConnection, crypter encrypt.Crypter) (*inco
 	if err != nil {
 		return nil, false, codedError(acctKeyErr, err)
 	}
-	return c.register(dc, nil, encKeyLegacy, acctPubB)
+	incompleteReg, paid, suspended, err := c.register(dc, nil, encKeyLegacy, acctPubB)
+	if suspended {
+		// TODO: This shouldn't happen since dc.acct.setupCryptoLegacy generates
+		// random keys, but perhaps reset the keys and re-generate new ones if
+		// this happens?
+		// Note that currently, the caller will think this account does not exist
+		// and attempt to pay the fee.
+		c.log.Infof("Legacy-style account key for %s was reported as suspended.", dc.acct.host)
+	}
```

Comment:
I'm leaning more and more towards not supporting DEXes that do not support HD accounts. It beats the purpose of using a seed if DEX operators do not support HD accounts.

Reply:
It's not unreasonable IMO either.  A server op could at least install the 0.2 patch that sends the dexpubkey if they don't want to update to 0.3.
Let's see what @buck54321 thinks about this.

Reply:
But regarding these lines of code, if `registerLegacy` finds the account is suspended, that should probably return and error to the caller for the reasons you cited (it's a fresh random identity).

Reply:
imo make everyone update, and don't support older dexes. I believe that is in line with the decred ethos, no soft forks.

Reply:
Cool, let's nuke `registerLegacy` and simplify `(*Core).Register` logic.

We'll still have to maintain some logic for imported accounts, but at least the registration bits won't have to worry about it.

Reply:
As long as users can still import accounts, OK. Really should be waiting for a major version change, but...

Reply:
True, and when we are at 1.0, I think the rules will change that way.  Although, right now 0.x means no promises.  Not that we should make a habit of making life difficult for users and operators, just that on the way to 1.0 we can be more forward-looking than we might otherwise be, otherwise we risk burning too much time and creating spaghetti to support obsolete features.
Definitely ack that POV though @buck54321.